internal/graph/graph.go

```diff
@@ -431,12 +433,22 @@ func NewLayeredCheckResolver(
 	localResolverOpts []LocalCheckerOption,
 	cacheEnabled bool,
 	cachedResolverOpts []CachedCheckResolverOpt,
+	throttlingEnabled bool,
+	throttler throttler.Throttler,
+	dispatchThrottlingCheckConfig *DispatchThrottlingCheckResolverConfig,
 ) (CheckResolver, CheckResolverCloser) {
 	cycleDetectionCheckResolver := NewCycleDetectionCheckResolver()
 	localCheckResolver := NewLocalChecker(localResolverOpts...)
 
 	cycleDetectionCheckResolver.SetDelegate(localCheckResolver)
 
+	var dispatchThrottlingCheckResolver *DispatchThrottlingCheckResolver
+	if throttlingEnabled {
+		dispatchThrottlingCheckResolver = NewDispatchThrottlingCheckResolver(dispatchThrottlingCheckConfig, throttler)
+		dispatchThrottlingCheckResolver.SetDelegate(localCheckResolver)
+		cycleDetectionCheckResolver.SetDelegate(dispatchThrottlingCheckResolver)
+	}
+
```

Comment:
The logic in line 455 below is incorrect if throttlingEnabled = true.   The cycleDetectionCheckResolver should not set delegate to cachedCheckResolver if we also have dispatchThrottlingCheckResolver.

Reply:
And this is why I suggested in https://github.com/openfga/openfga/pull/1571/files#r1588766760 moving this function to a separate file with unit tests. (we can do it in a separate PR

We need to unit test this better, there are too many combinations possible here and we can mess up easily.

Reply:
The construction logic is finished

Added a comment to address this in a seperate PR - https://github.com/openfga/openfga/pull/1571#issuecomment-2093143714